CRM/Financial/BAO/Payment.php

```diff
@@ -41,10 +42,13 @@ class CRM_Financial_BAO_Payment {
    * @throws \CRM_Core_Exception
    */
   public static function create(array $params): CRM_Financial_DAO_FinancialTrxn {
-    $contribution = civicrm_api3('Contribution', 'getsingle', ['id' => $params['contribution_id']]);
-    $contributionStatus = CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $contribution['contribution_status_id']);
+    $contribution = Contribution::get(FALSE)
+      ->addWhere('id', '=', $params['contribution_id'])
+      ->addSelect('*', 'contribution_status_id:name', 'balance_amount', 'paid_amount')
```

Comment:
Are 'balance_amount' & 'paid_amount' used?

Reply:
@colemanw not at this stage - but I put a couple of things in that were causing conflicts in other PRs when I tried to break it out into small chunks

Reply:
I'll let you keep at it then